.github/workflows/upgrade-testing.yml

```diff
@@ -0,0 +1,100 @@
+name: Pre/Post upgrade Testing
```

Comment:
This name feels a bit confusing. I suggest renaming it to "Migration Testing" or something similar, as it primarily handles tasks related to migration tests. However, we already have a "Migration Testing" in the state testing PR. Maybe we could scratch the Migration Testing workflow in the PR and keep this one, since this one is more comprehensive with all the customized inputs. What do you think?

Reply:
Yep, our end goal should be to completely remove the `migration-testing.yml` workflow and to use only this one from now on because it's much more customizable. I changed the naming because I wanted to de-couple the tests from the entire migration flow. If you insist, I'm good with reverting to the old namings (Migration Testing).

Reply:
Nice one yeah I would vote for the old name, Migration Testing.